docs/pull_request_template.md

```diff
@@ -0,0 +1,34 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Also, include motivation and context. List all dependencies required for this change (e.g. links to dependent PRs).
+
+Fixes # (issue)
+
+## Type of change
+
+Please mark options that are relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+# How To Test?
+
+Please describe the procedure and/or tests that need to be performed to verify these changes. Also, include test configuration details if relevant.
+
+- [ ] Your changes have to be tested on hardware
+- [ ] You have added unit tests
+
+## Test A
+1) Step 1
+2) Step 2
+
+## Test B
+1) Step 1
+2) Step 2
+
+# Checklist:
```

Comment:
@patapufate nit: 
```suggestion
# Checklist
```

Reply:
Also, shall we move this above testing procedures? It looks much more like a checklist for the author than anything else.

Reply:
Yes. Moved and fixed